config/config.go

```diff
@@ -223,6 +239,106 @@ func (c *Config) Validate() error {
 	return nil
 }
 
+// If the numerator in the Warp config is 0, use the default value
+func setQuorumNumerator(cfgNumerator uint64) uint64 {
+	if cfgNumerator == 0 {
+		return params.WarpDefaultQuorumNumerator
+	}
+	return cfgNumerator
+}
+
+// Helper to retrieve the Warp Quorum from the chain config.
+// Differentiates between subnet-evm and coreth RPC internally
+func getWarpQuorum(
+	subnetID ids.ID,
+	client ethclient.Client,
+) (WarpQuorum, error) {
+	if subnetID == constants.PrimaryNetworkID {
+		return WarpQuorum{
+			QuorumNumerator:   params.WarpDefaultQuorumNumerator,
+			QuorumDenominator: params.WarpQuorumDenominator,
+		}, nil
+	}
+
+	// Fetch the subnet's chain config
+	chainConfig, err := client.ChainConfig(context.Background())
+	if err != nil {
+		return WarpQuorum{}, fmt.Errorf("failed to fetch chain config for subnet %s: %v", subnetID, err)
+	}
+
+	// First, check the list of precompile upgrades to get the most up to date Warp config
+	upgrades := chainConfig.ToWithUpgradesJSON().UpgradeConfig.PrecompileUpgrades
+	for _, precompile := range upgrades {
+		warpConfig, ok := precompile.Config.(*warp.Config)
+		if ok {
+			return WarpQuorum{
+				QuorumNumerator:   setQuorumNumerator(warpConfig.QuorumNumerator),
+				QuorumDenominator: params.WarpQuorumDenominator,
+			}, nil
+		}
+	}
```

Comment:
Does this iterate from newest to oldest? If so, can you leave a comment here explaining that? We should make sure we get the newest warp config that exists.

Reply:
Added a check to keep only the most recent Warp config (if we find one at all in the upgrades config)